src/templates/directives/body.js

```diff
@@ -120,9 +120,11 @@ function addHyperlinks (text, onImgLoad, onImgClick) {
         text.addTemplateResult(
             url_obj.start,
             url_obj.end,
-            show_images && u.isImageURL(url_text) ?
-                u.convertToImageTag(url_text, onImgLoad, onImgClick) :
-                u.convertUrlToHyperlink(url_text),
+            show_images
+                && (u.isImageURL(url_text) || u.isWhitelistedImageURL(url_text))
+                && u.isImageDomainAllowed(url_text)
+                    ? u.convertToImageTag(url_text, onImgLoad, onImgClick)
+                    : u.convertUrlToHyperlink(url_text),
```

Comment:
`convertToImageTag` still has code that's no longer relevant due to the changes you've made here and in #2209.

Specifically the call to `u.isImageURL` and the check for `image_url_without_ext` are no longer relevant and can be removed.